src/parser/Clauses.h

```diff
@@ -115,6 +115,21 @@ class VertexIDList final {
         return result;
     }
 
+    void setContext(ExpressionContext *context) {
+        for (auto &expr : vidList_) {
+            expr->setContext(context);
+        }
+    }
+
+    std::vector<nebula::OptVariantType> eval() const {
+        std::vector<nebula::OptVariantType> vertices;
+        for (auto& vertex : vidList_) {
+            auto vid = vertex->eval();
```

Comment:
when vid is function expression, please do `vertex->prepare()` before `vid->eval()`

Reply:
Good Catch 👍